src/ccl_background.c

```diff
@@ -0,0 +1,1208 @@
+      ccl_check_status(cosmo,status);
+    }
+    else if(a[i]>1.) {
+      *status = CCL_ERROR_COMPUTECHI;
+      strcpy(cosmo->status_message,"ccl_background.c: scale factor cannot be larger than 1.\n");
+      ccl_check_status(cosmo,status);
+    }
+    else output[i]=5*log10(ccl_luminosity_distance(cosmo, a[i], status))+25;
+  }
+}
+
+//Scale factor for a given distance
+double ccl_scale_factor_of_chi(ccl_cosmology * cosmo, double chi, int * status)
+{
+  if((chi < 1.e-8) && (chi>=0.)) {
+    return 1.;
+  }
+  else if(chi<0.) {
+    *status = CCL_ERROR_COMPUTECHI;
+    strcpy(cosmo->status_message,"ccl_background.c: distance cannot be smaller than 0.\n");
+    ccl_check_status(cosmo,status);
+    return 0.;
+  }
+  else {
+    if (!cosmo->computed_distances) {
+      ccl_cosmology_compute_distances(cosmo,status);
+      ccl_check_status(cosmo,status);
+    }
+    double a;
+    int gslstatus = gsl_spline_eval_e(cosmo->data.achi, chi,cosmo->data.accelerator_achi, &a);
+    if(gslstatus != GSL_SUCCESS) {
+      ccl_raise_gsl_warning(gslstatus, "ccl_background.c: ccl_scale_factor_of_chi():");
+      *status |= gslstatus;
+    }
+    return a;
+  }
+}
+
+//
+void ccl_scale_factor_of_chis(ccl_cosmology * cosmo, int nchi, double chi[], double output[], int * status)
+{
+  if (!cosmo->computed_distances) {
+    ccl_cosmology_compute_distances(cosmo,status);
+    ccl_check_status(cosmo, status);    
+  }
+  int gslstatus;
+  for (int i=0; i<nchi; i++) {
+    if((chi[i] < 1.e-8) && (chi[i]>=0.))
+      output[i]=1.;
+    else if(chi[i]<0.) {
+      *status = CCL_ERROR_COMPUTECHI;
+      strcpy(cosmo->status_message,"ccl_background.c: distance cannot be less than 0.\n");
+      ccl_check_status(cosmo,status);
+    }
+    else
+      gslstatus = gsl_spline_eval_e(cosmo->data.achi,chi[i],cosmo->data.accelerator_achi,&output[i]);
+      if(gslstatus != GSL_SUCCESS) {
+        ccl_raise_gsl_warning(gslstatus, "ccl_background.c: ccl_scale_factor_of_chis():");
+        *status |= gslstatus;
+      }
+  }
+}
+
+double ccl_growth_factor(ccl_cosmology * cosmo, double a, int * status)
+{
+  if(a>1.) {
+    *status = CCL_ERROR_COMPUTECHI;
+    strcpy(cosmo->status_message,"ccl_background.c: scale factor cannot be larger than 1.\n");
+    ccl_check_status(cosmo,status);
+    return 0.;
+  }
+  else {
+    if (!cosmo->computed_growth) {
+      ccl_cosmology_compute_growth(cosmo, status);
+      ccl_check_status(cosmo, status);
+    }
+    if (*status!= CCL_ERROR_NOT_IMPLEMENTED) {
+      double D;
+      int gslstatus = gsl_spline_eval_e(cosmo->data.growth, a, cosmo->data.accelerator,&D);
+      if(gslstatus != GSL_SUCCESS) {
+        ccl_raise_gsl_warning(gslstatus, "ccl_background.c: ccl_growth_factor():");
+        *status |= gslstatus;
+        strcpy(cosmo->status_message, "ccl_background.c: ccl_growth_factor(): Scale factor outside interpolation range.\n");
+        return NAN;      
```

Comment:
So where do you sit on the "NaN corruption as a feature not a big" spectrum. My worry about returning 0 is that things are corrupted silently.

Reply:
:shrug: It's probably not that important, since the user (or python layer) should be checking the status anyway.  I was mostly advocating for being consistent.

Reply:
K. Let's go with NaN and see what happens.